pkg/build/controller/build/build_controller.go

```diff
@@ -326,7 +326,19 @@ func parseBuildConfigKey(key string) (string, string, error) {
 // handle function based on the build's current state. Each handler returns a buildUpdate
 // object that includes any updates that need to be made on the build.
 func (bc *BuildController) handleBuild(build *buildapi.Build) error {
-	if bc.shouldIgnore(build) {
+
+	// If pipeline build, handle pruning.
+	if build.Spec.Strategy.JenkinsPipelineStrategy != nil {
+		if buildutil.IsBuildComplete(build) {
+			if err := common.HandleBuildPruning(buildutil.ConfigNameForBuild(build), build.Namespace, bc.buildLister, bc.buildConfigGetter, bc.buildDeleter); err != nil {
+				utilruntime.HandleError(fmt.Errorf("failed to prune builds for %s/%s: %v", build.Namespace, build.Name, err))
+			}
+		}
+		glog.V(4).Infof("Ignoring build %s with jenkins pipeline strategy", buildDesc(build))
+		return nil
```

Comment:
probably shouldn't return here, in case at some point in the future we do *not* ignore pipeline builds below and want to actually do additional processing.